### PR TITLE
[#5105] Ongoing cumulative indicator period value

### LINKED ISSUE #5105: Save the ongoing cumulative indicator period value
### What are you trying to do?

The total value of period updates is saved as the `actual_value` attribute of the `period` object. The attribute used when generating the IATI export documents. Cumulative indicator updates should follow this process to work with other parts fo the system.

### Describe the solution you'd like

- [x] Save ongoing cumulative update values as the `period.actual_value` attribute

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zuhdil**
# TODO / Done

Summarize what has been changed / what has to be done in order to finalize the PR.

 - [x] Calculate cumulative update values as the period's actual values in a cumulative indicator
 - [x] Aggregate contributed cumulative period's actual values to parent period.
 - [x] Calculate cumulative indicator period's disaggregation values
 - [x] Aggregate contributed cumulative period disaggregation values to parent period.

# Test plan

What tests are necessary to ensure this works or doesn't break anything working

 - [x] Unit test
 
---

Closes #5105

### CODE REVIEW

File: akvo/rsr/tests/results_framework/test_cumulative_updates.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+                    {'period_start': self.PERIOD_2_START, 'period_end': date(2021, 12, 31)}
+                ]
+            }]
+        }])
+        if contributors:
+            builder.with_contributors(contributors)
+        return builder.build()
+
+
+class SingleUserCumulativeUnitUpdatesTestCase(BaseTestCase, CumulativeTestMixin):
```
Comment (MichaelAkvo) [inline akvo/rsr/tests/results_framework/test_cumulative_updates.py:38]: Mixins should come first due to [class resolution being from left to right](https://docs.python.org/3/tutorial/classes.html#multiple-inheritance).

**Review by MichaelAkvo — CHANGES_REQUESTED** [review]
Preliminary review. Have to look at the tests in depth

File: akvo/rsr/models/result/indicator.py [hunk 2]
```diff
@@ -201,6 +201,9 @@ def is_parent_indicator(self):
         """
         return self.child_indicators.count() > 0
 
+    def is_cumulative(self):
+        return self.cumulative and self.measure != PERCENTAGE_MEASURE
```
Comment (MichaelAkvo) [inline akvo/rsr/models/result/indicator.py:205]: Probably should add a comment to explain this

**Review by MichaelAkvo — CHANGES_REQUESTED** [review]
Just one comment to add. The rest looks fine

File: akvo/rsr/models/result/indicator_period.py [hunk 3]
```diff
@@ -172,30 +172,27 @@ def clean(self):
...
-                self.indicator.result.project.aggregate_children:
-            return self.recalculate_children(save)
-
+    def _calculate_cumulative_value(self):
+        '''
+        This method assumes the user will submit cumulative updates in chronological order as it should.
+        '''
+        value = 0
+        latest_per_users = get_per_user_latest_indicator_update_ids(self)
+        for update in IndicatorPeriodData.objects.filter(id__in=latest_per_users):
```
[the remark this answers is not in the document]
Comment (zuhdil) [reply akvo/rsr/models/result/indicator_period.py]: Fixed

File: akvo/rsr/tests/results_framework/test_cumulative_updates.py (hunk 1 above)
Comment (zuhdil) [reply akvo/rsr/tests/results_framework/test_cumulative_updates.py]: Fixed

File: akvo/rsr/models/result/indicator.py (hunk 2 above)
Comment (zuhdil) [reply akvo/rsr/models/result/indicator.py]: Fixed

**Pull request merged**